consensus/tests/miner.rs

```diff
@@ -71,4 +82,52 @@ mod miner {
         let parent_header = genesis().header.into();
         test_find_block(&transactions, &parent_header).await;
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn terminate_on_block() {
+        tracing_subscriber::fmt::init();
+        let consensus = snarkos_testing::sync::create_test_consensus().await;
+        consensus.fetch_memory_pool().await; // just make sure we're fully initialized by blocking on consensus call
+
+        let miner_address = FIXTURE_VK.test_accounts[0].address.clone();
+        let miner = MineContext::prepare(miner_address, consensus.clone()).await.unwrap();
+
+        let (sender, mut receiver) = mpsc::channel(10);
+
+        let terminator = Arc::new(AtomicBool::new(false));
+        let terminator_clone = terminator.clone();
+        tokio::task::spawn_blocking(move || {
+            let sender_clone = sender.clone();
+            let mining = async move {
```

Comment:
it probably works just fine, but I'm wondering if there was any reason for this `async` block to be put into the blocking task's scope, rather than before (and outside of) it?

Reply:
ah, ok - it's just a convenience block so it can be blocked on later